edge-util/src/Microsoft.Azure.Devices.Edge.Util/azureLogAnalytics/AzureLogAnalytics.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Util.AzureLogAnalytics
+{
+    using System;
+    using System.IO;
+    using System.Net;
+    using System.Security.Cryptography;
+    using System.Text;
+
+    /* Sample code from:
+    /* https://github.com/veyalla/MetricsCollector/blob/master/modules/MetricsCollector/AzureLogAnalytics.cs
+    /* https://dejanstojanovic.net/aspnet/2018/february/send-data-to-azure-log-analytics-from-c-code/
+    */
+
+    public class AzureLogAnalytics
+    {
+        public AzureLogAnalytics(string workspaceId, string sharedKey, string logType, string apiVersion = "2016-04-01")
+        {
+            this.WorkspaceId = workspaceId;
+            this.SharedKey = sharedKey;
+            this.LogType = logType;
+            this.ApiVersion = apiVersion;
+        }
+
+        public string WorkspaceId { get; }
+        public string SharedKey { get; }
+        public string ApiVersion { get; }
+        public string LogType { get; }
+
+        public void Post(byte[] content)
+        {
+            string requestUriString = $"https://{this.WorkspaceId}.ods.opinsights.azure.com/api/logs?api-version={this.ApiVersion}";
+            DateTime dateTime = DateTime.UtcNow;
+            string dateString = dateTime.ToString("r");
+            string signature = this.GetSignature("POST", content.Length, "application/json", dateString, "/api/logs");
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(requestUriString);
+            request.ContentType = "application/json";
+            request.Method = "POST";
+            request.Headers["Log-Type"] = this.LogType;
+            request.Headers["x-ms-date"] = dateString;
+            request.Headers["Authorization"] = signature;
+            using (Stream requestStreamAsync = request.GetRequestStream())
```

Comment:
GetRequestStream is not async method; it is better to use async method, try HttpClient (https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.getstreamasync?view=netframework-4.8).
```suggestion
            using (Stream requestStream = request.GetRequestStream())
```

Reply:
HttpClient